From 2da28dbd11e835ddc17e7656087b6f84c6f950ba Mon Sep 17 00:00:00 2001
From: George Kyriazis <george.kyriazis@intel.com>
Date: Wed, 9 Nov 2016 16:33:10 -0600
Subject: [PATCH 01/10] scons: ignore .hpp files in parse_source_list()

Drivers that contain C++ .hpp files need to ignore them too, along
with .h files, when building source file lists.

Reviewed-by: Emil Velikov <emil.velikov@collabora.com>
---
 scons/custom.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git scons/custom.py scons/custom.py
index bdb4039..544b15d 100644
--- scons/custom.py
+++ scons/custom.py
@@ -281,7 +281,7 @@ def parse_source_list(env, filename, names=None):
                     # cause duplicate actions.
                     f = f[len(cur_srcdir + '/'):]
                 # do not include any headers
-                if f.endswith('.h'):
+                if f.endswith(tuple(['.h','.hpp'])):
                     continue
                 srcs.append(f)
 
-- 
2.6.0

